webapp/channels/src/components/user_settings/display/user_settings_section.tsx

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable max-lines */
+
+import React from 'react';
+
+import {FormattedMessage} from 'react-intl';
+import {PrimitiveType, FormatXMLElementFn} from 'intl-messageformat';
+
+import SettingItem from 'components/setting_item';
+import SettingItemMax from 'components/setting_item_max';
+
+type ChildOption = {
+    id: string;
+    message: string;
+    value: string;
+    display: string;
+    moreId: string;
+    moreMessage: string;
+};
+
+type Option = {
+    value: string;
+    radionButtonText: {
+        id: string;
+        message: string;
+        moreId?: string;
+        moreMessage?: string;
+    };
+    childOption?: ChildOption;
+}
+
+type Props ={
+    section: string;
+    display: string;
+    defaultDisplay: string;
+    value: string;
+    title: {
+        id: string;
+        message: string;
+    };
+    firstOption: Option;
+    secondOption: Option;
+    thirdOption?: Option;
+    description: {
+        id: string;
+        message: string;
+        values?: Record<string, React.ReactNode | PrimitiveType | FormatXMLElementFn<React.ReactNode, React.ReactNode>>;
+    };
+    disabled?: boolean;
+    onSubmit?: () => void;
+    activeSection?: string;
+    handleSubmit: () => void
+    isSaving: boolean;
+    serverError?: string;
+    updateSection: (section: string) => void;
+    handleOnChange: (e: React.ChangeEvent, display: {[key: string]: any}) => void;
+}
+
+const UserSettingSection = (props: Props) => {
+    const {
+        section,
+        display,
+        value,
+        title,
+        firstOption,
+        secondOption,
+        thirdOption,
+        description,
+        disabled,
+        onSubmit,
+    } = props;
```

Comment:
I would prefer this expansion done in the previous line (inside the function arguments) to make sure the word `props` is not in the namespace and is not used.